test/desktop_sikuli/views/base_element.py

```diff
@@ -63,14 +76,40 @@ def verify_is_focused(self):
 
 class TextElement(object):
     def __init__(self, text):
+        self.text = text
         self.element_line = None
+
+    def find_element(self):
         for _ in range(3):
             lines = collectLines()
             for line in lines:
-                if text in line.getText().encode('ascii', 'ignore'):
+                if self.text in line.getText().encode('ascii', 'ignore'):
                     self.element_line = line
                     return
             time.sleep(3)
+        pytest.fail("Element with text '%s' was not found" % self.text)
 
     def click(self):
+        logging.info("Click %s button" % self.text)
+        self.find_element()
         self.element_line.click()
+
+    def get_whole_text(self):
+        self.find_element()
+        return self.element_line.getText().encode('ascii', 'ignore')
+
+    def is_visible(self):
+        from _pytest.runner import Failed
+        try:
+            self.find_element()
+            return True
+        except Failed:
+            return False
+
+    def verify_element_is_not_present(self):
+        from _pytest.runner import Failed
+        try:
+            self.find_element()
+        except Failed:
+            return
+        pytest.fail("'%s text is displayed but not expected'" % self.text)
```

Comment:
we're looking for an element and if found the method would fail with the message `...text is displayed but not expected`. otherwise, it would catch the exception and nothing return. is it correct?